api/src/services/notifications/update-service.ts

```diff
@@ -0,0 +1,29 @@
+import { Attributes } from "@sequelize/core"
+
+import { Notification } from "@/models"
+import BaseService from "@/services/base-service"
+import { isUndefined } from "lodash"
+
+export type NotificationUpdateAttributes = Partial<Attributes<Notification>>
+
+export class UpdateService extends BaseService {
+  constructor(
+    private notification: Notification,
+    private attributes: NotificationUpdateAttributes
+  ) {
+    super()
+  }
+
+  async perform() {
+    const { isRead } = this.attributes
+    // TODO: consider if isRead is a state change and should have it's own endpoint?
+    if (!isUndefined(isRead)) {
+      this.attributes.readDate = isRead === true ? new Date() : null
+    }
```

Comment:
See comment above about using a dedicated controller and treating this like a state change.
Given that it would remove this hard to read code, I'm liking it more and more.